### PR TITLE
Destroy webhook endpoints when user revokes oaurh app access

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
Destroy webhook endpoints when the user revokes oauth app access

## Related Tickets & Documents
#3715

### CODE REVIEW

File: app/controllers/oauth/tokens_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      #
+      # Doorkeeper does not use the token_type_hint logic described in the
+      # RFC 7009 due to the refresh token implementation that is a field in
+      # the access token model.
+      if authorized?
+        revoke_token
+        Webhook::Endpoint.where(user_id: token.resource_owner_id, oauth_application_id: token.application_id).destroy_all
+        render json: {}, status: :ok
+      else
+        error_description = I18n.t(:unauthorized, scope: %i[doorkeeper errors messages revoke])
```
[the remark this answers is not in the document]
Comment (maestromac) [reply app/controllers/oauth/tokens_controller.rb]: If the newer version is safe and we update to it, will there still be a need for this?

File: app/controllers/oauth/tokens_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    def revoke
+      # The authorization server, if applicable, first authenticates the client
+      # and checks its ownership of the provided token.
+      #
+      # Doorkeeper does not use the token_type_hint logic described in the
+      # RFC 7009 due to the refresh token implementation that is a field in
+      # the access token model.
+      if authorized?
+        revoke_token
+        Webhook::Endpoint.where(user_id: token.resource_owner_id, oauth_application_id: token.application_id).destroy_all
```
[the remark this answers is not in the document]
Comment (lightalloy) [reply app/controllers/oauth/tokens_controller.rb]: That makes sense, I'll make destroying async.

File: app/controllers/oauth/tokens_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (lightalloy) [reply app/controllers/oauth/tokens_controller.rb]: Here I'm overriding the action, so, ideally, we'll need to replace the action code with the new code (it'll be a bit more concise), but that's not required.

[the remark this answers is not in the document]
Comment (rhymes) [reply app/controllers/oauth/tokens_controller.rb]: I was thinking: what if we call "destroy if authorized?" and then call super? this way we delegate everything else to the `Doorkeeper::TokensController`

[the remark this answers is not in the document]
Comment (lightalloy) [reply app/controllers/oauth/tokens_controller.rb]: This way, if the `revoke_token` doesn't succeed for whatever reason, the webhook endpoints will still be destroyed.

**Pull request merged**